Applications/AdminUi/apps/admin_ui/lib/home/announcement_details/announcement_details.dart

```diff
@@ -17,28 +18,37 @@ class AnnouncementDetails extends StatefulWidget {
 }
 
 class _AnnouncementDetailsState extends State<AnnouncementDetails> {
-  Announcement? _announcmentDetails;
+  Announcement? _announcementDetails;
+
+  late final ScrollController _scrollController;
 
   @override
   void initState() {
     super.initState();
 
+    _scrollController = ScrollController();
+
     _loadAnnouncement();
   }
 
   @override
   void dispose() {
+    _scrollController.dispose();
     super.dispose();
   }
 
   @override
   Widget build(BuildContext context) {
-    if (_announcmentDetails == null) return const Center(child: CircularProgressIndicator());
+    if (_announcementDetails == null) return const Center(child: CircularProgressIndicator());
+
+    final announcementDetails = _announcementDetails!;
```

Comment:
```suggestion
    final announcementDetails = _announcementDetails;
    if (announcementDetails == null) return const Center(child: CircularProgressIndicator());
```

that should also work and is easier to read.